lib/generators/arqo/named_base.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails/generators/named_base'
+
+module ARQO
+  module Generators
+    class NamedBase < Rails::Generators::NamedBase
+      argument :attributes, type: :array, default: [], banner: 'associated_to[:class_name]'
+
+      private
+
+      def associated_to
+        attributes.find { |attribute| attribute.name == 'associated_to' }&.type
+      end
+
+      def associated_relation
+        associated_to&.to_s&.classify
+      end
+
+      def factory
+        (associated_to || class_name)&.to_s&.underscore
```

Comment:
can `associated_to` and  `class_name` be nil at the same time?

Reply:
Not really, `class_name` is always defined. 
Maybe safe operators are not necessary, are you asking because of that?

Reply:
exactly